sql/src/main/java/io/crate/executor/transport/ChainableActions.java

```diff
@@ -45,14 +47,15 @@
         ChainableAction<R> lastAction = actions.get(0);
         CompletableFuture<R> future = lastAction.doIt();
 
+        Result<R> result = new Result<>(null, null);
         for (int i = 1; i < actions.size(); i++) {
             ChainableAction<R> action = actions.get(i);
             previousActions.add(lastAction);
             lastAction = action;
-            future = future.handle(Result::new)
+            future = future.handle((r, f) -> Result.mergeWith(result, r, f))
```

Comment:
Would it make sense to have `mergeWith` as an instance method so that it would be possible to keep using method refs?